### DISCUSSION TITLE
file-logger 用log_format自定义日志内容时，有些变量打印不出来，比如$http,请求头等

### DISCUSSION THREAD

**Discussion by wangyang19930324** [apache/apisix#9814] [Q&A]
file-logger 用log_format自定义日志内容时，有些变量打印不出来，比如$http,请求头等，哪位大佬给解决一下

**Comment by jiangfucheng** [comment]
Can you provide corresponding configuration？

**Reply by wangyang19930324** [reply]
{
  "_meta": {
    "disable": false
  },
  "include_resp_body": true,
  "log_format": {
    "content.accessKey": "{请求秘钥}",
    "content.apiCode": "{接口编码}",
    "content.apiName": "{接口名称}",
    "content.args": "$args",
    "content.callId": "$request_id",
    "content.callResult": "{调用结果}",
    "content.eeeeee": "$request",
    "content.ffffff": "$http",
    "content.httpStatus": "$status",
    "content.proxyIp": "$balancer_ip",
    "content.proxyPort": "$balancer_port",
    "content.reqHeader": "$req_header",
    "content.reqIp": "$host",
    "content.reqTime": "$time_iso8601",
    "content.requestBody": "$request_body",
    "content.requestMethod": "$request_method",
    "content.respBody": "$resp_body",
    "content.routeId": "$route_id",
    "content.rspTime": "{响应时间}",
    "content.serviceCode": "{服务编码}",
    "content.serviceName": "$route_name",
    "content.timeConsume": "$request_time",
    "content.uri": "$uri",
    "content.userId": "$consumer_name",
    "logger": "apisixCallLog",
    "time": "$time_iso8601"
  },
  "path": "../logs/apisix-new.log"
}

**Reply by wangyang19930324** [reply]
$http  $args  $request_body  $req_header等都打印不出来

**Comment by kamly** [comment]
config:
```
{
    "plugins": {
        "file-logger": {
            "_meta": {
                "disable": false
            },
            "disable": false,
            "log_format": {
                "content.http": "$http",
                "content.httpStatus": "$status",
                "@timestamp": "$time_iso8601",
                "created_at": "$time_iso8601",
                "domain": "$host",
                "forward_ips": "$http_x_forwarded_for",
                "ip": "$http_x_real_ip",
                "method": "$request_method",
                "protocol": "$server_protocol",
                "referer": "$http_referer",
                "request": "$request",
                "request_time": "$request_time",
                "server_ip": "$server_addr",
                "size": "$body_bytes_sent",
                "status": "$status",
                "ua": "$http_user_agent",
                "upstream_addr": "$upstream_addr",
                "upstream_time": "$upstream_time",
                "uri": "$uri"
            },
            "path": "logs/file.log"
        }
    }
}
```

log
```
{"domain":"127.0.0.1","protocol":"HTTP/1.1","@timestamp":"2023-07-25T04:49:55+00:00","status":"502","request":"GET /pageview HTTP/1.1","method":"GET","uri":"/pageview","content.httpStatus":"502","created_at":"2023-07-25T04:49:55+00:00","upstream_addr":"127.0.0.1:8081","request_time":0.002,"size":"10","ua":"curl/7.74.0","server_ip":"127.0.0.1","route_id":"2"}
```
I didn’t encounter the situation you mentioned, and it can print `$http`, `$status` normally
You can try the configuration I provided, if you still encounter problems, you can see `errror.log` (./logs/error.log)
hope this helps
